### PR TITLE
Fix kubelet start failed on windows 10

### PR DESCRIPTION
**Pull request by llhhbc**
#### Which issue(s) this PR fixes:

Fixes kubelet start failed on Windows 10

image

#### Special notes for your reviewer:
kubelet-1.22

#### Does this PR introduce a user-facing change?
The sock that kubelet listen on windows is: `npipe:////./pipe/var/lib/kubelet/device-plugins/kubelet.sock`, and the sock that kubelet monitors on unix is: `/var/lib/kubelet/device-plugins/kubelet.sock`

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by llhhbc** [comment]
@matthyx OK

**Comment by bart0sh** [comment]
@llhhbc It's a user visible change in my opinion. Please, update release notes.

**Comment by llhhbc** [comment]
@matthyx  All test is ok.

**Comment by llhhbc** [comment]
@bart0sh Ok.

**Review by matthyx — APPROVED** [review]
/lgtm

**Comment by ffromani** [comment]
Obviously legit and important issue. similar to https://github.com/kubernetes/kubernetes/pull/115133 which is actively being discussed. Once sig-node and sig-windows converge on the preferred solution I'll comment here.

**Comment by ffromani** [comment]
Everything in the PR seems straightforward and LGTM bar the actual path construction inhttps://github.com/kubernetes/kubernetes/pull/113391/commits/44d35919fe4f9ab9926dfca1ed3fdaf8cb4f3ccc#diff-7f85a4e3a84eea0549eb8386e4c4efb40507c235d54acbb24ed85a1f0dcab58dR26 which is what I hinted previously regarding the similarities with https://github.com/kubernetes/kubernetes/pull/115133 . I'm not experienced enough in windows to comment, I can only say that containerd seems to use the forward slashes like unix (`//./.pipe/...`). Once we know what's the correct/preferred way to construct the path we can move forward and unblock.

**Comment by aravindhp** [comment]
@llhhbc what's the scenario where you see this issue? Is this specific to just Windows 10 or is this seen on Windows Server 2019 / 2022 also? @marosset @jsturtevant please take a look also.

**Comment by marosset** [comment]
The kubelet logs for one of the Windows nodes in this run says

> I0129 08:44:04.079801    4776 container_manager_windows.go:117] "Creating device plugin manager"
I0129 08:44:04.079801    4776 manager.go:125] "Creating Device Plugin manager" path="C:\\var\\lib\\kubelet\\device-plugins\\kubelet.sock"
I0129 08:44:04.079801    4776 server.go:65] "Creating device plugin registration server" version="v1beta1" socket="C:\\var\\lib\\kubelet\\device-plugins\\kubelet.sock"
I0129 08:44:04.081657    4776 fake_topology_manager.go:68] "AddHintProvider" hintProvider=&{checkpointdir:C:\var\lib\kubelet\device-plugins\ endpoints:map[] mutex:{state:0 sema:0} server:0xc0002e6310 activePods:0x259fde0 sourcesReady:0x7dfbcc8 allDevices:map[] healthyDevices:map[] unhealthyDevices:map[] allocatedDevices:map[] podDevices:0xc000e90160 checkpointManager:0xc000a23da0 numaNodes:[] topologyAffinityStore:0xc000a1e690 devicesToReuse:map[] pendingAdmissionPod:<nil>}

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/113391/pull-kubernetes-e2e-capz-windows-containerd/1619605598058319872/artifacts/clusters/capz-conf-tdnqzi/machines/capz-conf-tdnqzi-md-win-c495474f4-s958g/kubelet.log

I think the current behavior is correct and the Windows 10 client being used may not have proper support for unix domain sockets.

Given that we only claim to support (and test) Windows Server we shouldn't take this change as is.
cc @claudiubelu too

**Comment by jsturtevant** [comment]
I agree with @marosset the rest of the system relies on this being a UDS socket

**Comment by llhhbc** [comment]
Can this be merged?

**Comment by jsturtevant** [comment]
The rest of the device management code relies on UDS. It seems like windows 10 doesn't support UDS and we don't support running kubelet on windows 10 so this change can not merge as it will break device management on windows server.  

We don't officiall support windows 11 either but it may have the require apis. Have you tried running it there?

**Comment by llhhbc** [comment]
OK, Thanks. We have an old user system who needs to use windows 10 and cannot upgrade the system, so we can only temporarily modify it manually

**Comment by jsturtevant** [comment]
Thanks, As is this PR can't be excepted.  Let's open an issue if need further discussion
/close

**Pull request closed without merging**
